### PR TITLE
allow to use API token with Python tools

### PR DESCRIPTION
**Pull request by vladak**
This change broadens the HTTP header support in the Python tools. This is primarily done to allow Bearer API token to be used e.g. for `opengrok-sync` arguments and configuration.

For `opengrok-sync` specifically, this allows to specify the headers both as command line (`-H/--header` , accepts multiple values) and also in the configuration file, e.g.:
```yml
headers:
  'Authorization': 'Bearer foo'
  'X-Special-header': 'bar'
commands:
- command:
  ...
```

### CODE REVIEW

**Comment by vladak** [comment]
https://github.com/oracle/opengrok/wiki/Per-project-management-and-workflow will need update once this is in.

File: dev/before_install [hunk 1]
```diff
@@ -14,7 +14,6 @@ if [[ "$RUNNER_OS" == "Linux" ]]; then
 	    cssc \
 	    bzr \
 	    subversion \
-	    monotone \
```
Comment (vladak) [inline dev/before_install:17]: can't no longer install the package on Ubuntu

**Review by ahornace — APPROVED** [review]

**Comment by vladak** [comment]
While testing this on staging server I figured the need to allow token based authorization without the secure channel, so I added the `allowInsecureTokens` configuration tunable to the web app.

**Comment by vladak** [comment]
For reference, here is the `opengrok-sync` configuration I used to test this:

```yml
headers:
  'Authorization': 'Bearer foobar'
commands:
- command:
  - http://192.160.0.1:8080/source/api/v1/messages
  - POST
  - messageLevel: warning
    duration: PT1H
    tags: ['%PROJECT%']
    text: resync + reindex in progress
- command: [sudo, -u, wsmirror, /opengrok/dist/bin/venv/bin/opengrok-mirror, 
  -U, 'http://192.160.0.1:8080/source',
  -H, 'Authorization: Bearer foobar',
  -b, -c, /opengrok/etc/mirror-config-token.yml]
- command: [sudo, -u, webservd, /opengrok/dist/bin/venv/bin/opengrok-reindex-project,
    --java, '/jdk/openjdk11/bin/java',
    '-J=-XX:-UseGCOverheadLimit', -J=-Xmx16g, -J=-server, --jar, /opengrok/dist/lib/opengrok.jar,
    -H, 'Authorization: Bearer foobar',
    -U, 'http://192.160.0.1:8080/source',
    -t, /opengrok/etc/logging.properties.template, -p, '%PROJ%', -d, /opengrok/log/%PROJECT%,
    -P, '%PROJECT%', --, --renamedHistory, 'on', -r, dirbased, -G, -m, '256', --leadingWildCards, 'on',
    -c, /opengrok/bin/ctags, -U, 'http://192.160.0.1:8080/source', -o, /opengrok/etc/ctags.config,
    '--token', 'foobar',
    -H, '%PROJECT%']
  env: {LC_ALL: en_US.UTF-8}
  limits: {RLIMIT_NOFILE: 1024}
- command:
  - 'http://192.160.0.1:8080/source/api/v1/messages?tag=%PROJECT%'
  - DELETE
  - 'resync + reindex in progress'
  - 'Content-type': 'text/plain'
- command: [/scripts/check-indexer-logs.ksh]
cleanup:
- command:
  - 'http://192.160.0.1:8080/source/api/v1/messages?tag=%PROJECT%'
  - DELETE
  - 'resync + reindex in progress'
  - 'Content-type': 'text/plain'
```
Prior to that I had the same config but without the `headers`.

And prior to that I used another one that API commands lacked the header with the token, like so:
```yml
cleanup:
- command:
  - 'http://192.160.0.1:8080/source/api/v1/messages?tag=%PROJECT%'
  - DELETE
  - 'resync + reindex in progress'
  - 'Content-type': 'text/plain'
    'Authorization': 'Bearer foobar'
```
and `opengrok-sync` was run with `-H 'Authorization: Bearer foobar'`.

I do realize most of the header propagation is not covered by tests.

**Review by ahornace — APPROVED** [review]

**Comment by vladak** [comment]
thanks @ahornace !

Updated the relevant wikis.

**Pull request merged**